deno_dist/client/utils.ts

```diff
@@ -9,7 +9,7 @@ export const mergePath = (base: string, path: string) => {
 
 export const replaceUrlParam = (urlString: string, params: Record<string, string>) => {
   for (const [k, v] of Object.entries(params)) {
-    const reg = new RegExp('/:' + k + '({[^}]*})?')
+    const reg = new RegExp('/:' + k + '(\\{([^{}]*?(\\{[^{}]*?\\})?)+?\\})?')
```

Comment:
If we just pass that test pattern, the following simple code will work as expected.

```ts
const reg = new RegExp('/:' + k + '({[^/]+})?')
```

Reply:
I committed it including non capturing group. Please review it.